### PR TITLE
Fixup 3007

### PR DESCRIPTION
**Pull request by j9ac9k**
Re-add commented out tests in #3007

### CODE REVIEW

File: pyqtgraph/functions.py [hunk 1]
```diff
@@ -1257,7 +1257,7 @@ def rescaleData(data, scale, offset, dtype=None, clip=None):
         # Cupy does not support nditer
         # https://github.com/cupy/cupy/issues/5021
 
-        data_out = data.astype(work_dtype, copy=True)
+        data_out = data.astype(work_dtype)
```
Comment (pijyoi) [inline pyqtgraph/functions.py:1260]: I think `copy=True` is necessary as we are going to mutate the array in the next 2 statements

Comment (pijyoi) [reply pyqtgraph/functions.py:1260]: Oh I see, `copy=True` is the default

Comment (pijyoi) [reply pyqtgraph/functions.py:1260]: Oh, I remember why it was there. To make it symmetric to the `return data_out.astype(out_dtype, copy=False)`

Comment (j9ac9k) [reply pyqtgraph/functions.py:1260]: oh no..... I deleted the wrong `copy=True`

**Pull request merged**